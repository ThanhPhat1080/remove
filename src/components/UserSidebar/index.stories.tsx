// Lib
import { Story } from '@storybook/react';

// Components
import UserSidebar, { UserSidebarProps } from '@components/UserSidebar';

// Mocks
import { mockDataUser } from '@mocks/mockDataUser';

export default {
  title: 'UserSidebar',
  component: UserSidebar,
};

export const Sample: Story<UserSidebarProps> = () => <UserSidebar users={mockDataUser} onRemoveUser={() => {}} />;
