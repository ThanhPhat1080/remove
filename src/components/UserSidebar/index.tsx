// Lib
import { Link } from 'react-router-dom';

// Components
import ListUser from '@components/ListUser';
import Sidebar from '@components/Sidebar';

// Types
import { IUser } from 'types/user';

export type UserSidebarProps = {
  users: IUser[];
// @ts-ignore
onRemoveUser: (id: string) => void;
};

const UserSidebar = ({ users, onRemoveUser }: UserSidebarProps) => {
  return (
    <Sidebar>
      <h2 className='title sidebar-title'>List User</h2>
      <ListUser onRemoveUser={onRemoveUser} listUser={users} />
      <Link to={'/register'}>
        <button className='button'>New user</button>
      </Link>
    </Sidebar>
  );
};

export default UserSidebar;
