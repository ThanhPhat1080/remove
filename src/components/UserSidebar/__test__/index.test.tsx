// Lib
import { render, screen } from '@testing-library/react';

// Components
import UserSidebar, { UserSidebarProps } from '@components/UserSidebar';

// Mocks
import { mockDataUser } from '@mocks/mockDataUser';

describe('Test UserSidebar component', () => {
// @ts-ignore
  const initUserSidebar: UserSidebarProps = {
    users: mockDataUser,
  };

  test('It should match DOM Snapshot', () => {
    const { container } = render(<UserSidebar {...initUserSidebar} />);

    expect(container).toMatchSnapshot();
  });

  test('It should match data for UserSidebar component', () => {
    render(<UserSidebar {...initUserSidebar} />);

    expect(screen.getAllByRole('img')[0].getAttribute('src')).toEqual(
      'https://haycafe.vn/wp-content/uploads/2022/03/Avatar-hai-1.jpg',
    );

    expect(screen.getAllByRole('link')[0].textContent).toEqual('user1');
  });
});
