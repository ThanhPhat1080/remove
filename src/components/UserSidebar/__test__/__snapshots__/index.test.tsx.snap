// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test UserSidebar component It should match DOM Snapshot 1`] = `
<div>
  <div
    class="sidebar-wrapper"
  >
    <div>
      <h2
        class="title"
      >
        List User
      </h2>
      <ul
        class="list-item"
      >
        <li
          class="item"
        >
          <img
            alt="Avatar"
            class="avatar"
            src="https://haycafe.vn/wp-content/uploads/2022/03/Avatar-hai-1.jpg"
          />
          <a
            class="link"
            href="#"
          >
            user1
          </a>
          <img
            alt="delete"
            class="icon"
            src="IMAGE_MOCK"
          />
        </li>
        <li
          class="item"
        >
          <img
            alt="Avatar"
            class="avatar"
            src="https://tapchianhdep.com/wp-content/uploads/2021/05/hinh-avatar-hai-huoc-ngau.jpg"
          />
          <a
            class="link"
            href="#"
          >
            user2
          </a>
          <img
            alt="delete"
            class="icon"
            src="IMAGE_MOCK"
          />
        </li>
        <li
          class="item"
        >
          <img
            alt="Avatar"
            class="avatar"
            src="https://i.pinimg.com/736x/1c/44/41/1c444192c50fd9101640cdf99fd4a6c7.jpg"
          />
          <a
            class="link"
            href="#"
          >
            user3
          </a>
          <img
            alt="delete"
            class="icon"
            src="IMAGE_MOCK"
          />
        </li>
        <li
          class="item"
        >
          <img
            alt="Avatar"
            class="avatar"
            src="https://1.bp.blogspot.com/-Ada3siBi-Z8/YS3asSghKxI/AAAAAAABVSc/EJKXWlcxH1shPt6rWc-q3gxvSk2EdyZyACLcBGAsYHQ/s0/Avatar-con-viet-vang-ngau-Taihinhanh-Vn%2B%25286%2529.jpg"
          />
          <a
            class="link"
            href="#"
          >
            user4
          </a>
          <img
            alt="delete"
            class="icon"
            src="IMAGE_MOCK"
          />
        </li>
      </ul>
      <button
        class="button"
      >
        New user
      </button>
    </div>
  </div>
</div>
`;
